Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.h

```diff
@@ -121,9 +123,27 @@ class PDFIncrementalLoader : public ThreadSafeRefCountedAndCanMakeThreadSafeWeak
 
     Ref<PDFPluginStreamLoaderClient> m_streamLoaderClient;
 
+    class SemaphoreWrapper : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<SemaphoreWrapper> {
```

Comment:
👍 this is first time i saw this powerful class (I used Box<T> for my local test branch).

Reply:
Oh, I guess I could use Box too. I didn't realize Box used ThreadSafeRefCounted. I assumed it was regular RefCounted.

Reply:
Actually, my class is more convenient because it can be used with ThreadSafeWeakHashSet

Reply:
Agreed!